### PR TITLE
Updated docstring for find_constrained_prior

### PR DESCRIPTION
**Pull request by jlindbloom**
**What is this PR about?**

Just updated the `find_constrained_prior` docstring during the data umbrella sprint (with @bdatko).

#DataUmbrellaPyMCSprint
Cc: @reshamas

**Checklist**

+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Docs / Maintenance
- Just updated the `find_constrained_prior`, following [this guide](https://pymc-data-umbrella.xyz/en/latest/sprint/tutorials/docstring_tutorial.html#edit-the-docstring). 

## References
Towards #5459

### CODE REVIEW

**Comment by cluhmann** [comment]
@AlexAndorra you originally wrote this, correct?  Can you take a look?  It's for #DataUmbrellaPyMCSprint

File: pymc/func_utils.py [hunk 1]
```diff
@@ -50,24 +50,25 @@ def find_constrained_prior(
...
+        For instance, for a StudentT, you fix nu to a constant and get the optimized
         mu and sigma.
 
     Returns
     -------
-    The optimized distribution parameters as a dictionary with the parameters'
-    name as key and the optimized value as value.
+    opt_params: dict
+        The optimized distribution parameters as a dictionary with the parameters' and
+        name as key and the optimized value as value.
```
Comment (michaelosthege) [inline pymc/func_utils.py]:
```suggestion
        The optimized distribution parameters as a dictionary.
        Dictionary keys are the parameter names and
        dictionary values are the optimized parameter values.
```
The original phrasing was actually correct, but this alternative might be less confusing?

File: pymc/func_utils.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def find_constrained_prior(
 ) -> Dict[str, float]:
     """
     Find optimal parameters to get `mass` % of probability
-    of `pm_dist` between `lower` and `upper`.
+    of a `pm_dist` between `lower` and `upper`.
```
Comment (OriolAbril) [inline pymc/func_utils.py]:
```suggestion
    of a :ref:`distribution <api_distributions>` between `lower` and `upper`.
    
```

The distribution change is so the text shows distribution but rendered as a link to the distributions api page. The empty line is needed to comply with numpydoc: https://pymc-data-umbrella.xyz/en/latest/sprint/tutorials/docstring_tutorial.html#short-summary preferably one line, can't be more than one long sentence though and definotely not a whole paragraph with a note too

File: pymc/func_utils.py [hunk 3]
```diff
@@ -50,24 +50,26 @@ def find_constrained_prior(
         Lower bound to get `mass` % of probability of `pm_dist`.
```
Comment (OriolAbril) [inline pymc/func_utils.py:51]: In distribution parameter, use only "Distribution" to use the alias and have it rendered as a link in the docs. Alias source: https://github.com/pymc-devs/pymc/blob/main/docs/source/conf.py#L85

Comment (OriolAbril) [reply pymc/func_utils.py:51]: It is some lines above but I can't comment there

File: pymc/func_utils.py [hunk 4]
```diff
@@ -50,24 +50,26 @@ def find_constrained_prior(
...
     upper : float
         Upper bound to get `mass` % of probability of `pm_dist`.
-    init_guess: Dict[str, float]
+    init_guess : str or float
```
Comment (OriolAbril) [inline pymc/func_utils.py]: This is changing the type info of the parameter. It should be

```suggestion
    init_guess : dict of {str : float} 
```

Same for the other parameters. Ref: https://pandas.pydata.org/pandas-docs/stable/development/contributing_docstring.html#parameter-types

File: pymc/func_utils.py [hunk 5]
```diff
@@ -50,24 +50,26 @@ def find_constrained_prior(
...
         Initial guess for ``scipy.optimize.least_squares`` to find the
         optimal parameters of `pm_dist` fitting the interval constraint.
         Must be a dictionary with the name of the PyMC distribution's
         parameter as keys and the initial guess as values.
-    mass: float, default to 0.95
+    mass : float, default to 0.95
```
Comment (OriolAbril) [inline pymc/func_utils.py]:
```suggestion
    mass : float, default 0.95
```

File: pymc/func_utils.py [hunk 6]
```diff
@@ -50,24 +50,26 @@ def find_constrained_prior(
...
         Dictionary of fixed parameters, so that there are only 2 to optimize.
-        For instance, for a StudenT, you fix nu to a constant and get the optimized
+        For instance, for a StudentT, you fix nu to a constant and get the optimized
         mu and sigma.
 
     Returns
     -------
-    The optimized distribution parameters as a dictionary with the parameters'
-    name as key and the optimized value as value.
+    opt_params: dict
```
Comment (OriolAbril) [inline pymc/func_utils.py]: Adding the type is necessary and needed to be fixed. Giving a name to the returned value is optional, but if it is done, it must follow the same conventions as the parameters section

```suggestion
    opt_params : dict
```

Ref: https://pymc-data-umbrella.xyz/en/latest/sprint/tutorials/docstring_tutorial.html#returns-and-yields

**Comment by jlindbloom** [comment]
Thanks @OriolAbril, this is useful. I will reference these formatting changes moving forward.

File: pymc/func_utils.py [hunk 7]
```diff
@@ -36,38 +36,41 @@ def find_constrained_prior(
...
-    distribution : pm.Distribution
+    distribution : Distribution
         PyMC distribution you want to set a prior on.
         Needs to have a ``logcdf`` method implemented in PyMC.
     lower : float
         Lower bound to get `mass` % of probability of `pm_dist`.
     upper : float
         Upper bound to get `mass` % of probability of `pm_dist`.
-    init_guess: Dict[str, float]
+    init_guess : dict of {str : float}
```
Comment (cluhmann) [inline pymc/func_utils.py]: Just need to trim this trailing whitespace.

Comment (jlindbloom) [inline pymc/func_utils.py]:
```suggestion
    init_guess : dict of {str : float}
```

**Comment by cluhmann** [comment]
LGTM.  Just need a core dev to approve!

**Review by michaelosthege — APPROVED** [review]

**Comment by reshamas** [comment]
@jlindbloom 
PyMC office hours are on Thurs, Jul 14, in case there are any questions on this PR:
https://www.meetup.com/pymc-online-meetup/events/286864711/

**Comment by cluhmann** [comment]
I can't quite see what the whitespace CI failure is about.

**Comment by aloctavodia** [comment]
thanks @jlindbloom!

**Pull request merged**